modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/fragment/renderer/FDSAdminFragmentRenderer.java

```diff
@@ -450,16 +450,14 @@ private JSONObject _getCreationMenuJSONObject(
 				}));
 	}
 
-	private JSONObject _getDateJSONObject(Object object) {
-		if (object == null) {
+	private JSONObject _getDateJSONObject(String isoDate) {
+		if (isoDate == null) {
 			return null;
 		}
 
 		Calendar calendar = Calendar.getInstance();
 
-		Timestamp timestamp = (Timestamp)object;
-
-		calendar.setTime(new Date(timestamp.getTime()));
+		calendar.setTime(Date.from(Instant.parse(isoDate)));
```

Comment:
This was causing one of date range filter tests to fail, a Java error that object cannot be converted to a Timestamp. I don't know why this showed up now, it's 10 months old code. Did we already update Java version?